cpp/include/cudf/table/table_device_view.cuh

```diff
@@ -95,7 +95,10 @@ class mutable_table_device_view
  public:
   static auto create(mutable_table_view source_view, cudaStream_t stream = 0)
   {
-    auto deleter = [](mutable_table_device_view* t) { t->destroy(); };
+    auto deleter = [](mutable_table_device_view* t) {
```

Comment:
I think this should be done here https://github.com/rapidsai/cudf/blob/9951f71d4bc6f6d6fc5d5340bfa8e14b1a135738/cpp/src/table/table_device_view.cu#L29

And we need to handle this properly in table_device_view as well.

Reply:
Ram is right. The `destroy()` does a `delete this;` meaning the `t` variable should not be used after calling `t->destroy()`. The `delete _descendant_storage;` should go inside the `destroy()` method.

Reply:
Sounds good, let me take a look, I'll update shortly.

Reply:
@davidwendt @rgsl888prabhu please take another look. Thanks!

Reply:
@abellina run `python cpp/scripts/run-clang-format.py  -inplace` for style check to pass.

Reply:
@rgsl888prabhu done!